pkg/agents/team7/agent1/agent.go

```diff
@@ -79,99 +93,539 @@ func New(baseAgent *infra.Base) (infra.Agent, error) {
+		a.opMem.prevAge = a.Age()
+		a.opMem.msg1Sent = false
+		a.opMem.msg2Sent = false
+		a.opMem.msg3Sent = false
+	}
+
+	if !a.opMem.msg1Sent {
+		msg := messages.NewAskHPMessage(a.ID(), a.Floor(), a.Floor()+1)
+		// a.opMem.messageQueue = append(a.opMem.messageQueue, msg)
+		a.SendMessage(msg)
+		if a.Floor() != 1 {
+			msg = messages.NewAskHPMessage(a.ID(), a.Floor(), a.Floor()-1)
+			a.SendMessage(msg)
 		}
-		a.prevFloors = append(a.prevFloors, currentFloor)
+		a.opMem.msg1Sent = true
 	}
 
-	//receive Message
+	if a.PlatformOnFloor() && !a.opMem.msg2Sent {
+		if a.Floor() != 1 {
+			msg2 := messages.NewAskFoodTakenMessage(a.ID(), a.Floor(), a.Floor()-1)
+			a.SendMessage(msg2)
+		}
+		a.opMem.msg2Sent = true
+	}
+
+	if !a.PlatformOnFloor() && a.CurrPlatFood() != -1 && a.opMem.seenPlatform && a.Floor() != 1 && !a.opMem.msg3Sent {
+		msg2 := messages.NewAskFoodTakenMessage(a.ID(), a.Floor(), a.Floor()+1)
+		a.SendMessage(msg2)
+		a.opMem.msg3Sent = true
+	}
+
+}
+
+func (a *CustomAgent7) manageNewDay() {
 
-	//eat
+	a.opMem.seenPlatform = false
+	a.opMem.prevHP = a.HP()
+	a.opMem.foodEaten = 0
+	a.opMem.receivedReq = false
 
-	//choses random number between -5 and 5
-	if a.CurrPlatFood() != -1 {
+}
 
-		r1 := rand.Intn(11) - 5
-		r2 := rand.Intn(11) - 5
+func (a *CustomAgent7) treatyGT(tActive messages.Treaty, foodtotake food.FoodType, available food.FoodType, amount food.FoodType) food.FoodType {
+
+	if a.HP() > tActive.ConditionValue() || int(a.CurrPlatFood()) > tActive.ConditionValue() {
+		switch tActive.RequestOp() {
+		case messages.GT:
+			if foodtotake > available-food.FoodType(tActive.RequestValue()) {
+				foodtotake = available - food.FoodType(amount) - 1
+			}
+		case messages.GE:
+			if foodtotake >= available-food.FoodType(tActive.RequestValue()) {
+				foodtotake = available - food.FoodType(amount)
+			}
+		case messages.EQ:
+			if foodtotake != available-food.FoodType(tActive.RequestValue()) {
+				foodtotake = available - food.FoodType(amount)
+			}
+		}
+	}
+	return foodtotake
 
-		a.kindness += (a.personality.neuroticism * r1) / 50
-		a.greediness += (a.personality.neuroticism * r2) / 50
+}
+func (a *CustomAgent7) treatyGE(tActive messages.Treaty, foodtotake food.FoodType, available food.FoodType, amount food.FoodType) food.FoodType {
 
-		if a.kindness < 0 {
-			a.kindness = 0
-		} else if a.kindness > 100 {
-			a.kindness = 100
+	if a.HP() >= tActive.ConditionValue() || int(a.CurrPlatFood()) >= tActive.ConditionValue() {
+		switch tActive.RequestOp() {
+		case messages.GT:
+			if foodtotake > available-food.FoodType(tActive.RequestValue()) {
+				foodtotake = available - food.FoodType(amount) - 1
+			}
+		case messages.GE:
+			if foodtotake >= available-food.FoodType(tActive.RequestValue()) {
+				foodtotake = available - food.FoodType(amount)
+			}
+		case messages.EQ:
+			if foodtotake != available-food.FoodType(tActive.RequestValue()) {
+				foodtotake = available - food.FoodType(amount)
+			}
 		}
```

Comment:
You've copied this same block of code five times - you can put this into a function.